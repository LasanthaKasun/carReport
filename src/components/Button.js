import React from "react";

const Button = ({ text, onClick, role }) => {
  return (
    <button
      role={role}
      className={`${
        text !== "Back"
          ? "bg-blue-500 hover:bg-blue-700"
          : "bg-orange-500 hover:bg-orange-700"
      }text-white font-bold py-2 px-4 rounded`}
      onClick={onClick}
    >
      {text}
    </button>
  );
};

export default Button;
