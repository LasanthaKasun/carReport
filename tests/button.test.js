import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Button from "../src/components/Button";

it("should render button component", () => {
  const component = renderer.create(<Button />);
  expect(component).toMatchSnapshot();
});

it("should render button with props", () => {
  render(<Button text="Back" onClick={null} role="backBtn" />);
  const backBtn = screen.getByRole("backBtn");
  expect(backBtn).toHaveTextContent("Back");
});
