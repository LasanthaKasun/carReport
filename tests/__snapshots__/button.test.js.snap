// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render button component 1`] = `
<button
  className="bg-blue-500 hover:bg-blue-700text-white font-bold py-2 px-4 rounded"
/>
`;

exports[`should render error component 1`] = `
<button
  className="bg-blue-500 hover:bg-blue-700text-white font-bold py-2 px-4 rounded"
/>
`;
